Mathlib/CategoryTheory/FiberedCategory/BasedCategory.lean

```diff
@@ -0,0 +1,274 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.FiberedCategory.HomLift
+import Mathlib.CategoryTheory.Bicategory.Strict
+import Mathlib.CategoryTheory.Functor.Category
+
+/-!
+# The bicategory of based categories
+
+In this file we define the type `BasedCategory 𝒮`, and give it the structure of a strict
+bicategory. Given a category `𝒮`, we define the type `BasedCategory 𝒮` as the type of categories
+`𝒳` equiped with a functor `𝒳.p : 𝒳 ⥤ 𝒮`.
+
+We also define a type of functors between based categories `𝒳` and `𝒴`, which we call
+`BasedFunctor 𝒳 𝒴` and denote as `𝒳 ⥤ᵇ 𝒴`. These are defined as functors between the underlying
+categories `𝒳.obj` and `𝒴.obj` which commute with the projections to `𝒮`.
+
+Natural transformations between based functors `F G : 𝒳 ⥤ᵇ 𝒴 ` are given by the structure
+`BasedNatTrans F G`. These are defined as natural transformations `α` between the functors
+underlying `F` and `G` such that `α.app a` lifts `𝟙 S` whenever `𝒳.p.obj a = S`.
+-/
+
+universe v₅ u₅ v₄ u₄ v₃ u₃ v₂ u₂ v₁ u₁
+
+namespace CategoryTheory
+
+open CategoryTheory Functor Category NatTrans IsHomLift
+
+variable {𝒮 : Type u₁} [Category.{v₁} 𝒮]
+
+/-- A based category over `𝒮` is a category `𝒳` together with a functor `p : 𝒳 ⥤ 𝒮`. -/
+@[nolint checkUnivs]
+structure BasedCategory (𝒮 : Type u₁) [Category.{v₁} 𝒮] where
+  /-- The type of objects in a `BasedCategory`-/
+  obj : Type u₂
+  /-- The underlying category of a `BasedCategory`. -/
+  category : Category.{v₂} obj := by infer_instance
+  /-- The functor to the base. -/
+  p : obj ⥤ 𝒮
+
+instance (𝒳 : BasedCategory.{v₂, u₂} 𝒮) : Category 𝒳.obj := 𝒳.category
+
+/-- The based category associated to a functor `p : 𝒳 ⥤ 𝒮`. -/
+def BasedCategory.ofFunctor {𝒳 : Type u₂} [Category.{v₂} 𝒳] (p : 𝒳 ⥤ 𝒮) : BasedCategory 𝒮 where
+  obj := 𝒳
+  p := p
+
+/-- A functor between based categories is a functor between the underlying categories that commutes
+with the projections. -/
+structure BasedFunctor (𝒳 : BasedCategory.{v₂, u₂} 𝒮) (𝒴 : BasedCategory.{v₃, u₃} 𝒮) extends
+    CategoryTheory.Functor 𝒳.obj 𝒴.obj where
+  w : toFunctor ⋙ 𝒴.p = 𝒳.p := by aesop_cat
+
+/-- Notation for `BasedFunctor`. -/
+scoped infixr:26 " ⥤ᵇ " => BasedFunctor
+
+namespace BasedFunctor
+
+/-- The identity based functor. -/
+@[simps!]
```

Comment:
I believe it is better to generate a single simp lemma `toFunctor` rather than the projections `obj` and `map`:
```suggestion
@[simps toFunctor]
```
Same for `comp` below.